src/api.v2/helpers/pipeline/getPipelineStatus.js

```diff
@@ -0,0 +1,313 @@
+const _ = require('lodash');
+const AWS = require('../../../utils/requireAWS');
+
+const ExperimentExecution = require('../../model/ExperimentExecution');
+
+const config = require('../../../config');
+const getLogger = require('../../../utils/getLogger');
+const pipelineConstants = require('./constants');
+const { getPipelineStepNames } = require('./pipelineConstruct/skeletons');
+
+const logger = getLogger();
+
+// TODO: this lists should be computed from the actual state machines skeletons
+const qcPipelineSteps = [
+  'ClassifierFilter',
+  'CellSizeDistributionFilter',
+  'MitochondrialContentFilter',
+  'NumGenesVsNumUmisFilter',
+  'DoubletScoresFilter',
+  'DataIntegration',
+  'ConfigureEmbedding'];
+
+const gem2sPipelineSteps = [
+  'DownloadGem',
+  'PreProcessing',
+  'EmptyDrops',
+  'DoubletScores',
+  'CreateSeurat',
+  'PrepareExperiment',
+  'UploadToAWS'];
+
+// pipelineStepNames are the names of pipeline steps for which we
+// want to report the progress back to the user
+// does not include steps used to initialize the infrastructure (like pod deletion assignation)
+const pipelineSteps = getPipelineStepNames();
+
+// buildResponse function is wrapper function to ensure that all pipeline-status
+// responses contain the same information and parameters
+// more specific building response should rely on calling this one
+const buildResponse = (processName, execution, paramsHash, error, completedSteps) => {
+  const paramsHashOnlyIfGem2s = processName === pipelineConstants.GEM2S_PROCESS_NAME
+    ? paramsHash
+    : undefined;
+
```

Comment:
why was this addition necessary?

Reply:
For qc all `paramsHash` are `null` (it's is what we get from sql), so I was replacing it with `undefined`, but it might make more sense and be simpler to just keep it `null` honestly. I'll remove this code.

Reply:
Removed